b2sdk/transfer/emerge/emerger.py

```diff
@@ -40,6 +43,27 @@ def __init__(self, services):
         self.services = services
         self.emerge_executor = EmergeExecutor(services)
 
+    @staticmethod
+    def _update_file_info_with_large_file_sha1(
+        file_info: dict,
+        write_intents: List[WriteIntent],
+        emerge_plan: EmergePlan,
+        large_file_sha1: Optional[Sha1HexDigest] = None,
+    ) -> dict:
+        if not emerge_plan.is_large_file():
+            # Emerge plan doesn't construct a large file, no point setting the large_file_sha1
+            return file_info
+
+        if not large_file_sha1 and len(write_intents) == 1:
+            # large_file_sha1 was not given explicitly, but there's just one write intent, perhaps it has a hash
+            large_file_sha1 = write_intents[0].get_large_file_sha1()
+
+        if large_file_sha1:
+            file_info = file_info or {}
```

Comment:
I see what you mean here, but most of the time `file_info` will be coming from one of `Bucket`'s methods.  Now in all these methods `file_info` is an optional argument defaulting to `None`.  The docstrings also explains that it can be a `dict` or `None`.  

You're right about the dict reference.  I'll update the code to ensure that dicts passed in are not modified.

Does this make sense?